dist/expected.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import "github.com/gonum/quad"
+
+// ExpectedFixed numerically computes the expected value of the input function
+// under the probability distribution specified by the Quantiler. That
+// is, ExpectedFixed computes
+//
+//  ev ≈ E[f] = int_-inf^inf f(x) p(x) dx
+//
+// using the number of quadrature locations specified by evals. Concurrent
+// sets the number of allowed concurrent evaluations of f.
+//
+// ExpectedFixed is a wrapper around quad.Fixed. See the documentation for more
+// information.
+func ExpectedFixed(f func(float64) float64, q Quantiler, evals, concurrent int) (ev float64) {
```

Comment:
You're right that the easy to change argument is useless.

I don't think it's the same as Sort and SortStable. The sort code uses two different algorithms. This is exactly the same behavior, and will give exactly the same answer (up to rounding), it's only a matter if f is concurrently executable or not. For example, quadrature with a fixed number of points and quadrature with an adaptive number of points should be different functions because the algorithm is different.